sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1916,15 +1920,15 @@ def set_application(self, channel, appl_code):
         # Apply DataPathInit
         return self.xcvr_eeprom.write("%s_%d" % (consts.STAGED_CTRL_APPLY_DPINIT_FIELD, 0), channel)
 
-    def get_num_channels(self):
-        if self.get_module_type_abbreviation() == 'QSFP+C':
-            return 4
-        return self.NUM_CHANNELS
-
     def get_error_description(self):
         dp_state = self.get_datapath_state()
         conf_state = self.get_config_datapath_hostlane_status()
-        for lane in range(self.get_num_channels()):
+        for lane in range(self.NUM_CHANNELS):
+            name = "{}_{}_{}".format(consts.STAGED_CTRL_APSEL_FIELD, 0, lane + 1)
+            appl = self.xcvr_eeprom.read(name)
+            if (appl is None) or ((appl >> 4) == 0):
+                continue
```

Comment:
this api should return error state of datapath and module any time it is called. but looks like that is not the case.

Reply:
It does return the failure in the current application init sequence.